go/libraries/doltcore/table/editor/keyless_table_editor.go

```diff
@@ -343,30 +354,85 @@ func applyEdits(ctx context.Context, tbl *doltdb.Table, acc keylessEditAcc) (*do
 			return nil, err
 		}
 
-		var ok bool
+		oldv := v
 		if v.Empty() {
 			// row does not yet exist
 			v, ok, err = initializeCardinality(delta.val, delta.delta)
+
 		} else {
 			v, ok, err = modifyCardinalityWithDelta(v, delta.delta)
+
 		}
 		if err != nil {
 			return nil, err
 		}
 
+		indexOpsToUndo := make([]int, len(indexEds))
+		defer func() {
```

Comment:
Be wary of putting defers inside of for loops. They are called at function boundaries and not at block boundaries, so you have to use an anonymous function as the body of your for loop to get the desired behavior (or break out the for loop body into its own function).

Reply:
subtle! Are you saying that it's preferable to put it in a separate closure? or will it still perform correctly if I move it outside of the for loop and `Undo` all of `applyEdits` at the same time?

Reply:
After re-reading, I suppose we want the granularity of a different closure in this case

Reply:
Yep! Separate closure for the `for` body, or at least all of the logic following the defer up until the closing block of the `for` body.